src/v/config/validators.cc

```diff
@@ -67,4 +67,53 @@ validate_connection_rate(const std::vector<ss::sstring>& ips_with_limit) {
     return std::nullopt;
 }
 
+std::optional<std::pair<ss::sstring, int64_t>>
+parse_client_group_byte_rate_quota(const ss::sstring& raw_option) {
+    auto del_pos = raw_option.find(":");
+    if (del_pos == ss::sstring::npos || del_pos == raw_option.size() - 1) {
+        return std::nullopt;
+    }
+
+    auto client_group = raw_option.substr(0, del_pos);
+    auto rate_str = raw_option.substr(del_pos + 1);
+
+    std::pair<ss::sstring, int64_t> ans;
+    ans.first = client_group;
+    try {
+        ans.second = std::stoi(rate_str);
+    } catch (...) {
+        return std::nullopt;
+    }
+    return ans;
+}
+
+std::optional<ss::sstring> validate_client_groups_byte_rate_quota(
+  const std::vector<ss::sstring>& groups_with_limit) {
+    absl::node_hash_set<ss::sstring> groups_set;
+    for (const auto& group_and_limit : groups_with_limit) {
+        auto parsing_setting = parse_client_group_byte_rate_quota(
+          group_and_limit);
+        if (!parsing_setting) {
+            return fmt::format(
+              "Can not parse group rate limit {}", group_and_limit);
+        }
+
+        for (const auto& group : groups_set) {
+            if (
+              std::string_view(group).starts_with(
+                std::string_view(parsing_setting->first))
+              || std::string_view(parsing_setting->first)
+                   .starts_with(std::string_view(group))) {
+                return fmt::format(
```

Comment:
I think there's more of a convention to throw exceptions that contain error messages instead of a method that returns message or nullopt if no error